packages/core/src/split-layout/SplitLayout.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+  ElementType,
+  forwardRef,
+  ReactElement,
+  Children,
+  useEffect,
+} from "react";
+import { FlexLayout, FlexLayoutProps } from "../flex-layout";
+import { PolymorphicComponentPropWithRef, PolymorphicRef } from "../utils";
+
+export type SplitLayoutProps<T extends ElementType> =
+  PolymorphicComponentPropWithRef<
+    T,
+    {
+      /**
+       * Defines the default behavior for how flex items are laid out along the cross axis on the current line.
+       */
+      align?: FlexLayoutProps<ElementType>["align"];
+      /**
+       * Allow the items to wrap as needed, default is true.
+       */
+      wrap?: FlexLayoutProps<ElementType>["wrap"];
+      /**
+       * Controls the space between left and right items.
+       */
+      gap?: FlexLayoutProps<ElementType>["gap"];
+    }
+  >;
+
+type SplitLayoutComponent = <T extends ElementType = "div">(
+  props: SplitLayoutProps<T>
+) => ReactElement | null;
+
+export const SplitLayout: SplitLayoutComponent = forwardRef(
+  <T extends ElementType>(
+    { children, wrap = true, ...rest }: SplitLayoutProps<T>,
+    ref?: PolymorphicRef<T>
+  ) => {
+    const warnChildren = Children.toArray(children).length > 2;
```

Comment:
You can use Children.count for this. Also I'd maybe put this line inside the effect so it doesn't run in prod

Reply:
i think children.count would ignore conditional children